src/Components/DetailTable.tsx

```diff
@@ -0,0 +1,14 @@
+import { detailData } from '../Data/Detail'
+import { useParams } from 'react-router-dom';
+import { TypographyStyle, TypographyStyle2 } from './Style';
+export const DetailComponent: React.FC = () => {
+  const { id } = useParams<{ id: string }>();
+  const itemId = +id!;  //coverted nullable string into non-nullable number
```

Comment:
why is this needed /

Reply:
by the use of useParams it returns string but we need a number so i converted this, and id shouldn't null ever, that'swhy usse "!" this ,,